add game Update API

setung opened the pull request:
game을 수정하는 API 입니다.

Game Entity에 updateGame() 메소드를 만들었습니다.

setung committed:
add game Update API

yourhrh commented:
로컬에서 충돌문제 해결해주세요~ ㅎㅎ 
feature/game_update 브랜치에 develop을 머지한 이후에
충돌을 해결하시고 다시  Push 해주시면 됩니다

yourhrh commented on src/main/java/kr/couchcoding/tennis_together/controller/game/GameController.java:
```diff
@@ -8,14 +8,14 @@
 @RestController
 @RequiredArgsConstructor
 @RequestMapping("/games")
+@Transactional
```
컨트롤러에 Transactional을 쓰나요??? Service에서는 동작할텐데 컨트롤러에서는 동작하지 않는것으로 알고있습니다.

yourhrh replied:
https://dimdim.tistory.com/entry/Spring-MVC-Controller-%ED%81%B4%EB%9E%98%EC%8A%A4%EC%97%90-Transactional-%EC%82%AC%EC%9A%A9%EC%97%90-%EB%8C%80%ED%95%9C-%EC%82%BD%EC%A7%88

되게 하는방법도 있는데 일반적으로 사용하지 않아서..

setung replied:
아.. 실수했습니다.. 수정했습니다.

yourhrh commented on src/main/java/kr/couchcoding/tennis_together/controller/game/GameController.java:
```diff
@@ -40,4 +40,27 @@ public void postGame(@RequestBody PostGameDTO postGameDTO, Authentication authen
+    @PatchMapping("/{gameNo}")
+    public void updateGame(@PathVariable Long gameNo, @RequestBody PostGameDTO gameDTO, Authentication authentication) {
+        User user = ((User) authentication.getPrincipal());
+        Game game = gameService.findGameByGameNoAndGameCreator(gameNo, user);
```
47~ 64번줄의 내용을 Service로 옮기는게 좋아보입니다.

setung replied:
옮겼습니다~ 
저는 최대한 서비스에 엔터티를 넘기려고 했었는데요~
컨트롤러와 서비스에 코드를 작성하는 기준(?)을 살펴봐야 겠습니다.

yourhrh replied:
비즈니스 로직에 관련된건 Service에 있고 요청 & 응답 메세지에 관련된건 Controller에 있다고 보시면 됩니다 ㅎㅎ 
Entity 수정에 관련된 로직은 엥간해서는 서비스가 좋습니다.

setung committed:
Merge branch 'develop' into feature/game_update

# Conflicts:
#	src/main/java/kr/couchcoding/tennis_together/controller/game/GameController.java
#	src/main/java/kr/couchcoding/tennis_together/domain/game/dao/GameRepository.java
#	src/main/java/kr/couchcoding/tennis_together/domain/game/service/GameService.java

setung commented:
> 로컬에서 충돌문제 해결해주세요~ ㅎㅎ feature/game_update 브랜치에 develop을 머지한 이후에 충돌을 해결하시고 다시 Push 해주시면 됩니다

네 수정했습니다~

setung committed:
Reflect review

 - move Transactional in Controller to Service
 - move Update logic in Controller to Service
